source/code/troubleshooter/omsagent_tsg_zip/tsg_files/tsg_code/connect/tsg_checkendpts.py

```diff
@@ -0,0 +1,82 @@
+# INSPIRED BY update_mgmt_health_check.py
+
+import socket
+
+from tsg_info             import tsginfo_lookup
+from tsg_errors           import tsg_error_info
+
+omsadmin_path = "/etc/opt/microsoft/omsagent/conf/omsadmin.conf"
+
+# ping specific endpoint
```

Comment:
any chance to add something above a ping test? e.g. a `curl` call or some other way to test for actual connection success. i think in many cases ping will succeed but that just says that the destination is reachable; if an actual connection is attempted it may be denied by any firewall rules the customer has applied.

Reply:
Sorry the comment here was misleading, it's actually a socket test, updated the comment here:
https://github.com/microsoft/OMS-Agent-for-Linux/blob/bff7c80f869d6f5a4046ed41b4bd0ed89d69219f/source/code/troubleshooter/modules/connect/check_endpts.py#L11
But I also have plans to improve this test by also using SSL checks and the certs, but that's for a future release